src/goto-programs/goto_program.cpp

```diff
@@ -634,3 +634,40 @@ void goto_programt::compute_incoming_edges()
   }
 }
 
+bool goto_programt::instructiont::equal(const instructiont &other) const
+{
+  return type == other.type && code == other.code && guard == other.guard &&
+         targets.size() == other.targets.size() && labels == other.labels;
+}
+
+bool goto_programt::equal(const goto_programt &other) const
+{
+  if(instructions.size() != other.instructions.size())
+    return false;
+
+  goto_programt::const_targett other_it = other.instructions.begin();
+  for(const auto &ins : instructions)
+  {
+    if(!ins.equal(*other_it))
+      return false;
+
+    long jump_difference1 = 0;
+    if(!ins.targets.empty())
```

Comment:
Should compare other targets I think -- e.g. `catch` instructions have several targets.